File: 5-LongestPalindromicSubstring/5-LongestPalindromicSubstring.py
# Last updated: 08/05/2026, 09:34:14
class Solution:
    def longestPalindrome(self, s: str) -> str:
        
        long=''
        def expand_center(left, right):
            while left >= 0 and right < len(s) and s[left] == s[right]:
                left -= 1
                right += 1
            return s[left + 1:right]

        i=0
        while i<len(s):

            #odd
            str1= expand_center(i,i)
            #even
            str2=expand_center(i,i+1)

            if len(long)<len(str1):
                long= str1
            if len(long)<len(str2):
                long= str2
            i+=1
        return long

File: 5-LongestPalindromicSubstring/5-LongestPalindromicSubstring.cpp
// Last updated: 08/05/2026, 09:34:08
class Solution {
public:
    string longestPalindrome(string s) {
        int n= s.size();
        int l;
        int r;
        int maxlen=0;
        string ans;
        for(int i=0;i<n;i++){
            l=i;
            r=i;
            //odd
            while(l>=0 && r<n && s[l]==s[r] ){
                if(r-l+1>maxlen){
                    maxlen=r-l+1;
                    ans=s.substr(l,maxlen);
                }
                l-=1;
                r+=1;
            }
            //even
            l=i;
            r=i+1;
            while(l>=0 && r<n && s[l]==s[r] ){
                if(r-l+1>maxlen){
                    maxlen=r-l+1;
                    ans=s.substr(l,maxlen);
                }
                l-=1;
                r+=1;
            }

        }
        return ans;
        
    }
};